src/kOS.Safe/Encapsulation/ScalarValue.cs

```diff
@@ -0,0 +1,456 @@
+﻿using System;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Exceptions;
+
+namespace kOS.Safe.Encapsulation
+{
+    abstract public class ScalarValue : Structure, IConvertible
+    {
+        protected object internalValue;
+
+        abstract public bool IsInt { get; }
+
+        abstract public bool IsDouble { get; }
+
+        public bool IsValid
+        {
+            get
+            {
+                if (IsInt) return true;
+                if (IsDouble)
+                {
+                    double d = GetDoubleValue();
+                    if (double.IsInfinity(d) || double.IsNaN(d)) return false;
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        public object Value { get { return internalValue; } }
+
+        protected ScalarValue()
+        {
+            InitializeSuffixes();
+        }
+
+        public void InitializeSuffixes()
+        {
+            AddSuffix("ISINT", new Suffix<bool>(() => IsInt));
+            AddSuffix("ISDOUBLE", new Suffix<bool>(() => IsDouble));
+            AddSuffix("ISVALID", new Suffix<bool>(() => IsValid));
+        }
+
+        public static ScalarValue Create(object value)
+        {
+            if (value is float)
+                value = Convert.ToDouble(value);
+            if (value is double)
+            {
+                bool inBounds = Int32.MinValue < (double)value && (double)value < Int32.MaxValue;
+                if (inBounds && !double.IsNaN((double)value))
+                {
+                    // Convert the double to an int, and check and see if they are still equal.
+                    // if so, treat the double as if it was an int.
+                    int intPart = Convert.ToInt32(value);
+                    if ((double)value == intPart)
```

Comment:
Agreed.  But until we get it set up that we can run basic scripts outside of kOS (a stand alone interpreter) we cannot easily use the debugger to confirm this is the case.  I'm of the mindset to leave these checks in until we know for sure that we have caught every method of instantiating.